src/requests/completions.jl

```diff
@@ -63,106 +65,50 @@ function latex_completions(doc, offset, partial, CIs)
-        if startswith("macro", spartial)
-            push!(CIs, CompletionItem("macro", 14, "macro", TextEdit(Range(doc, offset:offset), "macro \$1(\$2)\n    \$0\nend"[length(spartial) + 1:end])))
-        end
-        if startswith("module", spartial)
-            push!(CIs, CompletionItem("module", 14, "module", TextEdit(Range(doc, offset:offset), "module \$0\nend"[length(spartial) + 1:end])))
-        end
-        if startswith("mutable", spartial)
-            push!(CIs, CompletionItem("mutable", 14, "mutable", TextEdit(Range(doc, offset:offset), "mutable struct \$1\n    \$0\nend"[length(spartial) + 1:end])))
-        end
-    elseif startswith("outer", spartial)
-        push!(CIs, CompletionItem("outer", 14, "outer", TextEdit(Range(doc, offset:offset), "outer"[length(spartial) + 1:end])))
-    elseif startswith("primitive", spartial)
-        push!(CIs, CompletionItem("primitive", 14, "primitive", TextEdit(Range(doc, offset:offset), "primitive type \$1\n    \$0\nend"[length(spartial) + 1:end])))
-    elseif startswith("quote", spartial)
-        push!(CIs, CompletionItem("quote", 14, "quote", TextEdit(Range(doc, offset:offset), "quote\n    \$0\nend"[length(spartial) + 1:end])))
-    elseif startswith("return", spartial)
-        push!(CIs, CompletionItem("return", 14, "return", TextEdit(Range(doc, offset:offset), "return \$0"[length(spartial) + 1:end])))
-    elseif startswith("struct", spartial)
-        push!(CIs, CompletionItem("struct", 14, "struct", TextEdit(Range(doc, offset:offset), "struct \$1\n    \$0\nend"[length(spartial) + 1:end])))
-    elseif fc == 't'
-        if startswith("try", spartial)
-            push!(CIs, CompletionItem("try", 14, "try", TextEdit(Range(doc, offset:offset), "try \$1\n    \$0\ncatch\nend"[length(spartial) + 1:end])))
-        end
-    elseif startswith("using", spartial)
-        push!(CIs, CompletionItem("using", 14, "using", TextEdit(Range(doc, offset:offset), "using \$0\n"[length(spartial) + 1:end])))
-    elseif startswith("while", spartial)
-        push!(CIs, CompletionItem("while", 14, "while", TextEdit(Range(doc, offset:offset), "while \$1\n    \$0\nend"[length(spartial) + 1:end])))
     end
 end
 
+const snippet_completions = Dict{String,String}(
+    "abstract" => "abstract type \$0 end",
+    "baremodule" => "baremodule \$0\nend",
+    "begin" => "begin\$0\nend", 
+    "break" => "break",
+    "catch" => "catch",
+    "const" => "const ",
+    "continue" => "continue",
+    "do" => "do \$0\nend",
+    "else" => "else",
+    "elseif" => "elseif ",
+    "end" => "end",
+    "export" => "export ",
+    "finally" => "finally",
+    "for" => "for \$0 in \$1\nend",
+    "function" => "function \$0(\$1)\nend",
+    "global" => "global ",
+    "if" => "if \$0\nend",
+    "import" => "import",
+    "let" => "let \$0\nend",
+    "local" => "local ",
+    "macro" => "macro \$0(\$1)\nend",
+    "module" => "module \$0\nend",
+    "mutable" => "mutable struct \$0\nend",
+    "outer" => "outer ",
```

Comment:
Did `outer` have some sort of special meaning before 1.0? Or is there a form using `outer` that I'm not aware of?

```
❯ julia --lisp
;  _
; |_ _ _ |_ _ |  . _ _
; | (-||||_(_)|__|_)|_)
;-------------------|----------------------------------------------------------

> reserved-words
(begin while if for try return break continue function macro quote let local
 global const do struct module baremodule using import export end else elseif
 catch finally true false)
```

Reply:
It soft-scopes (is that a term?) the iterator of a loop to the parent (local) scope such that:
```julia
let
    i = 0
    for outer i = 1:10
    end
    @assert i == 10
end
```